64-blue-whale/technical_specification.md

```diff
@@ -0,0 +1,130 @@
+# Reimplement frontend features in Angular (Blue Whale)
+
+**Epic Type:** Implementation Epic
+
+Epic planning and implementation follow the
+[Epic Planning and Marathon SOP](https://docs.ghga-dev.de/main/sops/sop001_epic_planning.html).
+
+## Scope
+
+### Outline:
+
+The goal of this epic is to implement all the features from the current React based GHGA Data Portal in Angular, using the baseline application that has been built in the [Hippopotamus](../60-hippopotamus/technical_specification.md) as the starting point.
+
+### Included/Required:
+
+- Full implementation of all features of the React based GHGA Data Portal
+- Unit tests and end-to-end tests (against the mock backend) for these features
+
+### Optional:
+
+- Some features or change requests proposed for the existing GHGA Data Portal
+- Full test coverage
+
+### Not included:
+
+- Automated end-to-end tests against the real backend
+
+## Details:
+
+This section outlines a checklist of all features to be implemented, divided into several steps. A pre-release should be published upon the completion of each step. Final releases should begin with version 2 to distinguish them from the legacy frontend, which uses version numbers in the version 1 range.
+
+### Already implemented
+
+The following parts are already contained in the baseline application that has been created in the [Hippopotamus](../60-hippopotamus/technical_specification.md) epic.
+
+- Homepage
+	- Global Stats
+- Header and Footer
+- Dataset browser (partial)
+- Auth Service
+- Error message handling
+- User Login:
+	- Login/Logout buttons
+	- OIDC callback URL
+	- User registration
+	- 2FA setup
+	- 2FA confirmation
+- Metadata service
+- Metadata search service (partial)
+
+We have also added rudimentary unit tests and end-to-end tests. While implementing the remaining functionality in the following four steps, basic tests should be included together with the implemented functionality. As a last step, the implementation and tests should be refined.
+
+### Step 1: Catalog functionality
+
+In this epic, we first implement functions from the GHGA Catalog.
+
+- [ ] Metadata search service (complete)
+- [ ] Dataset browser (complete)
+	- [ ] Result list
```

Comment:
What is there left for the results list itself? The summary is the thing missing no?

Reply:
That's true, you did that already including pagination. Maybe we should add some tests and a loading indicator or ghost elements while waiting for results, because this can sometimes be slow.

Reply:
Adapted this.